pkg/geoipupdate/config.go

```diff
@@ -118,12 +132,21 @@ func NewConfig(
 		Parallelism:       1,
 	}
 
-	// Override config with values from the config file.
-	err := setConfigFromFile(config, path)
+	// Potentially populate config.configFilePath. We will rerun this function
+	// again later to ensure the flag values override env variables.
+	err := setConfigFromFlags(config, flagOptions...)
```

Comment:
Admittedly this is a bit strange: we apply the flag options here at the start, and again a few blocks lower. I wasn't sure how else to allow a new optional flag to be passed in that itself influenced behavior that we normally do before processing those flags. I'm not sure if this is the ideal way to handle this.